cmd/snap-update-ns/utils.go

```diff
@@ -286,6 +291,63 @@ func secureMkfileAll(name string, perm os.FileMode, uid, gid int) error {
 	return err
 }
 
+func planWritableMimic(dir string) ([]*Change, error) {
```

Comment:
I think a comment above here describing what the function does and why would be handy (eg, describing that the algorithm is intentionally simple so that the changes can be undone be simple going backwards, justifying mounting over 'dir' if you decide to keep the current implementation (eg, so the undo works as expected, there isn't much in there, etc), ....

Reply:
Done